Objective-C/Objective-C.sublime-syntax

```diff
@@ -13,200 +15,334 @@ first_line_match: |-
-  non_func_keywords: 'if|for|switch|while|decltype|sizeof|__declspec|__attribute__'
-
 contexts:
   main:
-    - include: preprocessor-global
-    - include: global
+    - meta_prepend: true
+    - include: interface
+    - include: implementation
+
+  implementation:
+    - match: ^\s*(@implementation)\s*({{identifier}})\b
+      captures:
+        1: meta.implementation.objc keyword.declaration.class.objc
+        2: meta.implementation.objc entity.name.type.objc
+      push: [implementation.meta, interface.body]
+  implementation.meta:
+    - meta_scope: meta.implementation.objc meta.scope.implementation.objc
+    - include: pop-immediately
+
+  interface:
+    - match: ^\s*(@interface|@protocol)\s*({{identifier}})\b
+      captures:
+        1: meta.interface-or-protocol.objc keyword.declaration.class.objc
+        2: meta.interface-or-protocol.objc entity.name.type.objc
+      push: [interface.meta, interface.body, interface.struct, interface.subclass, interface.category]
+  interface.meta:
+    - meta_scope: meta.interface-or-protocol.objc meta.scope.interface.objc
+    - include: pop-immediately
+  interface.category:
+    - clear_scopes: 1
+    - match: \(
+      push: enclosure.parenthesis
+    - include: else-pop
+  interface.subclass:
+    - clear_scopes: 1
+    - include: protocol-list
+    - match: ':'
+      scope: punctuation.separator.objc
+    - include: expression-identifiers
+    - include: else-pop
+  interface.struct:
+    - match: \{
+      scope: punctuation.section.block.begin.objc
+      set: enclosure.block
+    - include: else-pop
+  interface.body:
+    - match: \@end
+      scope: keyword.declaration.interface.end.objc
+      pop: 1
+    - match: \@property
+      scope: keyword.other.objc
+      push: [statement, property]
+    - match: \@(?:synthesize|dynamic)
+      scope: keyword.other.objc
+      push: statement
 
-  #############################################################################
-  # Reusable contexts
-  #
-  # The follow contexts are currently constructed to be reused in the
-  # Objective-C++ syntax. They are specifically constructed to not push into
-  # sub-contexts, which ensures that Objective-C++ code isn't accidentally
-  # lexed as plain Objective-C.
-  #
-  # The "unique-*" contexts are additions that Objective-C makes over C, and
-  # thus can be directly reused in Objective-C++ along with contexts from C++
-  # and C.
-  #############################################################################
+    - match: (?:\+|\-)\s*(?=\()
+      push: [selector, selector.return]
+    - include: expression
 
-  properties:
-    - match: ((@)property)\s*(\()
+  property:
+    - match: \(
+      scope: punctuation.section.group.begin.objc
+      push: property.attrs
```

Comment:
If `@property ()()()` is not a valid statement, I'd suggest to `set` property.attr.

```suggestion
      set: property.attrs
```